driver/normalizer/normalizer.go

```diff
@@ -27,7 +28,78 @@ var Preprocessors = []Mapping{
 	}.Mapping(),
 }
 
+func mapIASTNameDerived(typ string) Mapping {
+	return MapSemantic(typ, uast.Identifier{}, MapObj(
+		Obj{
+			"Name": Var("name"),
+		},
+		Obj{
+			"Name": Var("name"),
+		},
+	))
+}
+
+type opJoinNamesArray struct {
+	qualified Op
+}
+
+func (op opJoinNamesArray) Kinds() nodes.Kind {
+	return nodes.KindString
+}
+
+func (op opJoinNamesArray) Check(st *State, n nodes.Node) (bool, error) {
+	res, err := op.qualified.Check(st, n)
```

Comment:
It wasn't clear to me where to put the reverse transformation but until confirming with you I suspected it was there :) I'll add it.

Reply:
I just commited this. Please check because I'm not sure I did everything as it should and I also don't know how to check the result of the reverse operation.

Reply:
Looks correct, thanks! The `uast.ToNode` is slow, in general, but since we won't use reverse that often, I think it's okay.

The best way is to write a small test for it. Should be something like:
```golang
op := opJoinNamesArray{Var("x")}

st := NewState()
n2, err := op.Check(st, n1)
// test that n2 should is a specific string

// reverse
n3, err := op.Construct(st, n2)
// test that it returned an array of objects with names
```

Reply:
Trying to get a test to work but getting very odd results (even in the case where it clearly works on the fixtures) :/